### PR TITLE
3.next - Response methods (file, cookie, download)

### PR DESCRIPTION
**Pull request by markstory**
Implement the last set of PSR-7-like methods. This is the last batch of this kind of work 🎉 

Refs #9636

### CODE REVIEW

File: src/Network/Response.php [hunk 1]
```diff
@@ -2078,6 +2144,104 @@ public function file($path, array $options = [])
...
+        if (!is_file($path)) {
+            $path = APP . $path;
+        }
+
+        $file = new File($path);
+        if (!$file->exists() || !$file->readable()) {
+            if (Configure::read('debug')) {
+                throw new NotFoundException(sprintf('The requested file %s was not found or not readable', $path));
+            }
+            throw new NotFoundException(__d('cake', 'The requested file was not found'));
```
Comment (lorenzo) [inline src/Network/Response.php]: Didn't we get rid of error message translations?

Comment (ADmad) [inline src/Network/Response.php]: We did for developer centric message, this would be shown to end user.

Comment (dereuromark) [inline src/Network/Response.php]: Sure? I thought those are only visible with debug true.

Comment (markstory) [inline src/Network/Response.php]: This is a non-debug message.

Comment (lorenzo) [inline src/Network/Response.php]: A few lines up there is a message that is by shown in debug without translations. I'm more inclined to not have translations at all for those messages

Comment (inoas) [inline src/Network/Response.php]: The non-debug messages should be translated imho. Else it is required to hack to core to have a non-english software towards end users.

File: src/Network/Response.php [hunk 2]
```diff
@@ -2078,6 +2144,104 @@ public function file($path, array $options = [])
...
+     * Validate a file path is a valid response body.
+     *
+     * @param string $path The path to the file.
+     * @throws \Cake\Network\Exception\NotFoundException
+     * @return \Cake\Filesystem\File
+     */
+    protected function validateFile($path)
+    {
+        if (strpos($path, '../') !== false || strpos($path, '..\\') !== false) {
+            throw new NotFoundException('The requested file contains `..` and will not be read.');
```
Comment (inoas) [inline src/Network/Response.php]: Misses if `(Configure::read('debug')) {`? or needs to be translated?

**Comment by lorenzo** [comment]
Great job @markstory!

**Comment by markstory** [comment]
Thanks @lorenzo I'm going to do some quick regression tests and then start the beta releases for 3.4.0

**Pull request merged**